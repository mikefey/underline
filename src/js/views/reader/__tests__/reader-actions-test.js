import test from 'ava';
import {
  loadBook,
  updateCurrentBookData,
  updateBookPositionData,
  sliderPositionChanged,
  sliderReleased,
} from './../reader-actions';
import bookFixture from './../../../../../test/fixtures/book';

test('loadBook action', (t) => {
  t.snapshot(loadBook(1));
});

test('updateCurrentBookData action', (t) => {
  t.snapshot(updateCurrentBookData(bookFixture));
});

test('updateBookPositionData action', (t) => {
  t.snapshot(updateBookPositionData(1, { percentage: 0.76 }));
});

test('sliderPositionChanged action', (t) => {
  t.snapshot(sliderPositionChanged(0.56));
});

test('sliderReleased action', (t) => {
  t.snapshot(sliderReleased(0.65));
});
