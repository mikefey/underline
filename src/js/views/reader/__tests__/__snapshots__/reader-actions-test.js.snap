// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadBook action 1`] = `
Object {
  "id": 1,
  "type": "LOAD_BOOK",
}
`;

exports[`sliderPositionChanged action 1`] = `
Object {
  "percentage": 0.56,
  "type": "SLIDER_POSITION_CHANGED",
}
`;

exports[`sliderReleased action 1`] = `
Object {
  "percentage": 0.65,
  "type": "SLIDER_RELEASED",
}
`;

exports[`updateBookPositionData action 1`] = `
Object {
  "id": 1,
  "location": Object {
    "percentage": 0.76,
  },
  "type": "UPDATE_BOOK_POSITION_DATA",
}
`;

exports[`updateCurrentBookData action 1`] = `
Object {
  "data": Object {
    "author": "The book author",
    "cover_image": "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7",
    "current_location": Object {
      "end": null,
      "percentage": 0,
      "start": null,
    },
    "file": "blob",
    "id": 1,
    "parsed_data": null,
    "title": "A Good Book",
    "total_locations": 0,
  },
  "type": "UPDATE_CURRENT_BOOK_DATA",
}
`;
